src/p384/short_address.rs

```diff
@@ -0,0 +1,198 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * (c) ZeroTier, Inc.
+ * https://www.zerotier.com/
+ */
+use crate::p384::*;
+
+/// 128-bit short address prefix.
+///
+/// Short addresses are primarily for cases where humans need to type addresses or where
+/// they need to be mapped onto an IPv6 address. The fully qualified 384-bit address
+/// should be preferred if address transfer is automated or via cut/paste.
+#[repr(transparent)]
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub struct ShortAddress(pub(crate) [u64; 2]); // treated as [u8; 16]
+
+impl ShortAddress {
+    pub const SIZE: usize = 16;
+    pub const STRING_SIZE_NO_PREFIX: usize = 31;
+    pub const STRING_SIZE: usize = Self::STRING_SIZE_NO_PREFIX + PREFIX_SHORT.len();
+
+    #[inline(always)]
+    pub fn as_bytes(&self) -> &[u8; Self::SIZE] {
+        debug_assert_eq!(size_of::<[u8; Self::SIZE]>(), size_of::<Self>());
+        unsafe { &*(&self.0 as *const [u64; 2]).cast() }
+    }
+
+    #[inline(always)]
+    fn as_mut_bytes(&mut self) -> &mut [u8; Self::SIZE] {
+        debug_assert_eq!(size_of::<[u8; Self::SIZE]>(), size_of::<Self>());
+        unsafe { &mut *(&mut self.0 as *mut [u64; 2]).cast() }
+    }
+
+    #[inline(always)]
+    fn is_valid(&self) -> bool {
+        self.as_bytes()[0] == Address::REQUIRED_PREFIX
+    }
+
+    pub fn write_to_string(&self, s: &mut String, prefix: bool) {
+        if prefix {
+            s.push_str(PREFIX_SHORT);
+        }
+        first_128_to_string(self.as_bytes(), s);
+    }
+}
+
+impl TryFrom<[u8; 16]> for ShortAddress {
+    type Error = InvalidParameterError;
+
+    #[inline]
+    fn try_from(value: [u8; 16]) -> Result<Self, Self::Error> {
```

Comment:
Like the full-length Address above, I think we should be consistent about using the const size for the byte array (potentially via a type alias).

Reply:
Done!